src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -261,8 +270,8 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(viewContext));
             }
 
-            var action =
-                _urlHelper.RouteUrl(routeName, values: routeValues, protocol: null, host: null, fragment: null);
+            var urlHelper = _urlHelperFactory.Create(viewContext);
```

Comment:
Hmm, I see. because we don't capture `ViewContext` on the `DefaultHtmlGenerator`. Seems odd since we typically will have 1 `DefaultHtmlGenerator` per request.